### PR TITLE
Reorganise staking manager to allow for upgrades

### PR DESCRIPTION
**Pull request by drinkcoffee**
Reorganise staking manager code such that libraries that might be extended are abstract contracts with storage gaps.

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env sh
 . "$(dirname -- "$0")/_/husky.sh"
 
-export NVM_DIR="$HOME/.nvm"
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+# export NVM_DIR="$HOME/.nvm"
```
Comment (rzmahmood) [inline .husky/pre-commit:4]: Why is this commented out?

Comment (drinkcoffee) [reply .husky/pre-commit:4]: The precommit hook always fails with error 3. The recommended way of fixing the issue is by disabling those lines. Altering permissions of nvm doesn't appear to work.

File: contracts/lib/ChildManagerLib.sol [hunk 2]
```diff
@@ -1,27 +0,0 @@
-// SPDX-License-Identifier: MIT
-pragma solidity 0.8.19;
```
Comment (rzmahmood) [inline contracts/lib/ChildManagerLib.sol:2]: Why was this deleted?

**Review by rzmahmood — APPROVED** [review]
